### PR TITLE
[WIP][DX] Better exception page

### PR DESCRIPTION
**Pull request by hason**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? |  |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |
- [x] add twig files into stack trace
- [x] add syntax highlighters for PHP and Twig with symfony.com's color scheme
- [ ] use VarDumper for function arguments (maybe in another PR)
- [ ] add tests

**Before:**
before

**After:**
after

### CODE REVIEW

File: src/Symfony/Bundle/TwigBundle/Resources/views/Exception/traces.html.twig [hunk 1]
```diff
@@ -17,7 +17,7 @@
     <a id="traces-link-{{ position }}"></a>
     <ol class="traces list-exception" id="traces-{{ position }}" style="display: {{ 0 == count ? 'block' : 'none' }}">
```
Comment (jrobeson) [inline src/Symfony/Bundle/TwigBundle/Resources/views/Exception/traces.html.twig]: Can you use the same twig cycle technique you used below for this also?

File: src/Symfony/Bundle/TwigBundle/Resources/views/Exception/trace.html.twig [hunk 2]
```diff
@@ -1,3 +1,13 @@
+<div>
+{% spaceless %}
+    <span class="toggle-icon">
+        <a href="#" onclick="toggle('data-{{ prefix ~ '-' ~ i }}'); switchIcons('icon-{{ prefix ~ '-' ~ i }}-open', 'icon-{{ prefix ~ '-' ~ i }}-close'); return false;">
```
Comment (jrobeson) [inline src/Symfony/Bundle/TwigBundle/Resources/views/Exception/trace.html.twig]: if/when this gets merged, will people mind a PR that switches out the onclick methods and such for the class here that can be handled in in script tag included in the main layout file?  That way I don't have to keep full copies of these templates simply to please my CSP policy.

Comment (weaverryan) [inline src/Symfony/Bundle/TwigBundle/Resources/views/Exception/trace.html.twig]: @jrobeson I don't know if I understand. Also, this is just a movement of this code - we're already using the onclick like this (search for `toggle(` on this page - you'll see the part is was moved from)

**Comment by weaverryan** [comment]
@hason do you have some time to update this PR for the suggested changes? I'd still like to see this, e.g. for 2.8.

Thanks!

**Comment by fabpot** [comment]
@hason Do you time to finish this one? Or perhaps @javiereguiluz can take over?

**Comment by hason** [comment]
Closing in favour of #15792.

File: src/Symfony/Component/Debug/ExceptionProcessor.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+     */
+    public function flatten(\Exception $exception, $options = array())
+    {
+        $options = array_merge(array(
+            'class' => 'Symfony\Component\Debug\Exception\FlattenException',
+        ), $options);
+
+        $e = call_user_func(array($options['class'], 'create'), $exception);
+        foreach ($this->flatteners as $flattener) {
+            $e = $flattener->flatten($exception, $e, $options);
```
Comment (hason) [inline src/Symfony/Component/Debug/ExceptionProcessor.php]: Solved in #15792.

**Pull request closed without merging**